### PR TITLE
Relay peer datagrams as Data indications

### PR DESCRIPTION
**Pull request by erszcz**
Addresses #16 finishing the basic (i.e. non-channel) relay mechanism.

### CODE REVIEW

File: lib/fennec/udp/worker.ex [hunk 1]
```diff
@@ -92,10 +91,27 @@ defmodule Fennec.UDP.Worker do
...
-  def handle_peer_data(:allowed, _ip, _port, _data, state) do
+  defp data_params(ip, port, data) do
+    alias Jerboa.Params, as: P
+    alias Jerboa.Format.Body.Attribute.{Data, XORPeerAddress}
+    import Fennec.Evaluator.Helper, only: [family: 1]
+    P.new()
+    |> P.put_class(:indication)
+    |> P.put_method(:data)
+    |> P.put_attr(%Data{content: data})
+    |> P.put_attr(%XORPeerAddress{
```
Comment (rslota) [inline lib/fennec/udp/worker.ex]: As far as I know there is `new` constructor for all `XORAddress` structs that fills `family` for you. It's seems like a better way to go. It has been added to `Jerboa` very recently, so the `fennec` code didn't have a chance to adapt yet.

File: test/fennec/udp/data_test.exs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    test "gets discarded when there's no permission for peer", ctx do
+      ## given a relay address for a peer with no permission
+      %XORRelayedAddress{address: relay_addr, port: relay_port} = ctx.relay_sock
+      {:ok, peer} = :gen_udp.open(0, [{:active, :false}, :binary])
+      {:ok, peer_port} = :inet.port(peer)
+      self_ = self()
+      with_mock Fennec.UDP.Worker, [:passthrough], [
+        handle_peer_data: fn (:no_permission, ip, port, data, state) ->
+          ## we can't use `assert called ...` as we want to ignore `state`
```
Comment (rslota) [inline test/fennec/udp/data_test.exs]: What about `assert called Fennec.UDP.Worker.handle_peer_data(@peer_addr, ^peer_port, ^data, :_)`?

Comment (erszcz) [inline test/fennec/udp/data_test.exs]: `assert called ...` only calls `:meck.called(...)` - it's not a macro, but a function call, therefore it requires a proper value (not a wildcard) for each of the passed in arguments.

Comment (rslota) [inline test/fennec/udp/data_test.exs]: Yes, that's all true. But `:_` is an atom that is interpreted by `meck` as wildcard, so whether it's macro or function doesn't matter :)

**Pull request merged**